### ISSUE TITLE
Parquet file database

### ISSUE THREAD

**Issue by skyhirider** [zencq/Pi#9]
Hi, I've converted the data you extracted into parquet files.

Compared to the original 10GB, the size of the repo is 848 MB.

The data can be opened in VS Code and queries like a database with the correct parquet reader.

See https://github.com/skyhirider/nms-technology-upgrade-database

**Comment by cengelha** [comment]
Hey, that looks interesting! I'll keep the csv files nonetheless as it should be human readable and useable by everyone (some may not know about VSC and/or parquet files or how to use it, install extensions, etc, etc).

But it might be interesting for the integration in NN (smaller download size/faster processing)!

**Comment by skyhirider** [comment]
@cengelha yes, could be an optional separate download, or if you focus on the seeds that are good and skip subpart seeds the size can be significantly reduced as well.

It's something I was missing in the java based save editor I was using for checking mods, had to copy paste the seeds and check in a separate table.

Would be cool if the save editor showed that the mod is number 51 best. Could make the list smaller for download too if it was only an ordered list

**Comment by cengelha** [comment]
That always bothered me too and was the reason why I stared the Pi repo (always in mind adding it to NN eventually).

For completeness I would like to always keep all seeds but the size reduction by over 90% alone is already huge. 

My plan is to not include all files by default but to make each one individually downloadable when requested (either auto. when opening the details or manually with a button in it; probably both via setting).

When downloaded (or already exists) a new list will be show (or replacing the current one; not sure yet) to show all seeds and its stats. The list should be highlight the current seed and be searchable and sortable.
There might be a potential performance issue when loading 10k records so I may be forced to skip some seeds anyway (I have to try and see).

**Comment by cengelha** [comment]
@skyhirider Hey, I just pushed the first updates including parquet files. So every item has now also a parquet file additionally to the csv file :) 
They are all created with the default Snappy compression.

**Comment by skyhirider** [comment]
Nice! I saw the release, thanks for updating the seeds.

If possible, could you bundle the parquets into its own archive?

I'm thinking a zip file under releases that would have parquets for the best mods (s for exocraft, x for the rest bar a few exceptions) would be the one that would be most desired by people and shou be a small download.

**Comment by cengelha** [comment]
I'll think about the release stuff after catching up with the latest release.

For the best seeds there is the [Pi.xlsx](https://github.com/zencq/Pi/blob/main/Pi.xlsx) that already has them. Includes S und S/X for those that have both. Every item shows the seeds with the best of each stat and the overall best. The screenshot I posted yesterday on Discord is from that file.
Gets generated with the [generate_xlsx.py](https://github.com/zencq/Pi/blob/main/generate_xlsx.py) by reading and analyzing all the current files.
